diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf98289..7316268 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,7 +85,9 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     SET_PROPERTY(CACHE ENABLE_SIMD_FLAGS PROPERTY STRINGS none native SSE3 SSE4.2 AVX AVX2) 
 
     if(${ENABLE_SIMD_FLAGS} MATCHES "native")
-     	add_definitions(-march=native -mfpmath=both)
+        # The fpmath option is x86 only.
+        # add_definitions(-march=native -mfpmath=both)
+        add_definitions(-march=native)
         message(STATUS "Enabling native optimizations")
     elseif(${ENABLE_SIMD_FLAGS} MATCHES "AVX2")
     	add_definitions(-mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mpopcnt -mavx -mfma -mavx2 -mfpmath=both)
diff --git a/SoapyLMS7/Settings.cpp b/SoapyLMS7/Settings.cpp
index 4de1ea0..bfd68df 100644
--- a/SoapyLMS7/Settings.cpp
+++ b/SoapyLMS7/Settings.cpp
@@ -169,7 +169,7 @@ SoapySDR::Kwargs SoapyLMS7::getHardwareInfo(void) const
     info["protocolVersion"] = devinfo.protocolVersion;
     if (devinfo.boardSerialNumber != unsigned(-1))
     {
-        char buff[64]; sprintf(buff, "0x%lx", devinfo.boardSerialNumber);
+        char buff[64]; sprintf(buff, "0x%llx", devinfo.boardSerialNumber);
         info["boardSerialNumber"] = buff;
     }
     return info;
diff --git a/src/lms7002m/LMS7002M_RxTxCalibrations.cpp b/src/lms7002m/LMS7002M_RxTxCalibrations.cpp
index 05ed415..3b02fb2 100644
--- a/src/lms7002m/LMS7002M_RxTxCalibrations.cpp
+++ b/src/lms7002m/LMS7002M_RxTxCalibrations.cpp
@@ -1022,7 +1022,7 @@ int LMS7002M::CalibrateTx(float_type bandwidth_Hz, bool useExtLoopback)
     Log("Setup stage", LOG_INFO);
     status = CalibrateTxSetup(bandwidth_Hz, useExtLoopback);
 #ifdef __cplusplus
-    verbose_printf("Setup duration: %li ms\n",
+    verbose_printf("Setup duration: %lli ms\n",
         std::chrono::duration_cast<std::chrono::milliseconds>
         (std::chrono::high_resolution_clock::now()-beginTime).count());
 #endif
diff --git a/src/lms7002m_mcu/MCU_BD.cpp b/src/lms7002m_mcu/MCU_BD.cpp
index 98eb9d8..5095ec9 100644
--- a/src/lms7002m_mcu/MCU_BD.cpp
+++ b/src/lms7002m_mcu/MCU_BD.cpp
@@ -1032,7 +1032,7 @@ int MCU_BD::WaitForMCU(uint32_t timeout_ms)
     }while (std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count() < timeout_ms);
     mSPI_write(0x0006, 0); //return SPI control to PC
     //if((value & 0x7f) != 0)
-    lime::debug("MCU algorithm time: %li ms", std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count());
+    lime::debug("MCU algorithm time: %lli ms", std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count());
     return value & 0x7F;
 }
 
