diff --git a/radio/blocks/init.lua b/radio/blocks/init.lua
index ff14250..c58f3f4 100644
--- a/radio/blocks/init.lua
+++ b/radio/blocks/init.lua
@@ -18,6 +18,7 @@ return {
     PulseAudioSource = require('radio.blocks.sources.pulseaudio'),
     SoapySDRSource = require('radio.blocks.sources.soapysdr'),
     UHDSource = require('radio.blocks.sources.uhd'),
+    ShortTextMessageSource = require('radio.blocks.sources.shorttextmessage'),
 
     -- Sink Blocks
     IQFileSink = require('radio.blocks.sinks.iqfile'),
@@ -37,6 +38,7 @@ return {
     SoapySDRSink = require('radio.blocks.sinks.soapysdr'),
     NopSink = require('radio.blocks.sinks.nop'),
     UHDSink = require('radio.blocks.sinks.uhd'),
+    ShortTextMessageSink = require('radio.blocks.sinks.shorttextmessage'),
 
     -- Signal Blocks
     --- Filtering
@@ -81,6 +83,7 @@ return {
     SamplerBlock = require('radio.blocks.signal.sampler'),
     SlicerBlock = require('radio.blocks.signal.slicer'),
     DifferentialDecoderBlock = require('radio.blocks.signal.differentialdecoder'),
+    ManchesterEncoderBlock = require('radio.blocks.signal.manchesterencoder'),
     ManchesterDecoderBlock = require('radio.blocks.signal.manchesterdecoder'),
     --- Type Conversion
     ComplexToRealBlock = require('radio.blocks.signal.complextoreal'),
@@ -108,4 +111,7 @@ return {
     POCSAGDecoderBlock = require('radio.blocks.protocol.pocsagdecoder'),
     --- Varicode
     VaricodeDecoderBlock = require('radio.blocks.protocol.varicodedecoder'),
+    --- Simple framer/deframer
+    SimpleFramerBlock = require('radio.blocks.protocol.simpleframer'),
+    SimpleDeframerBlock = require('radio.blocks.protocol.simpledeframer'),
 }
